test/functional/wallet_taproot.py

```diff
@@ -387,6 +388,14 @@ def do_test_psbt(self, comment, pattern, privmap, treefn, keys_pay, keys_change)
         psbt_online.unloadwallet()
         psbt_offline.unloadwallet()
 
+    def do_test_k_of_n(self, k, n):
```

Comment:
@Sjors Given my basic understanding of output descriptors, I'd much appreciate your thoughts on this block of code, to make sure it meets the issue criterias and if this is a good way to test it.

Reply:
I understand descriptors, but Python less well :-) What is this function trying to do?

Reply:
It's a function that allows setup and testing of any k-of-n wallet. It builds on the idea of the existing test that @darosior pointed to, which starts on L495 https://github.com/bitcoin/bitcoin/blob/64440bb733896a7a2caf902825e0406cb993e666/test/functional/wallet_taproot.py#L495 where a random number of `n` is used.

The part that differs is how the `[KEY_1,...,KEY_N]` array inside the `multi_a` is built - where the previous function uses `([H_POINT] * rnd_pos)` up until `k1` and then `([H_POINT] * (MAX_PUBKEYS_PER_MULTI_A - 1 - rnd_pos)` after, the new function uses `k1` for the whole array of.

However, because this previous test is not documented, I'm not entirely sure about the H_POINT parts do and if I need them in my test somehow, or if the way I've written it is enough to verify the 999-of-999 and 1-of-1000 cases.